src/test/java/com/specialwarriors/conal/GithubRepoServiceTest.java

```diff
@@ -0,0 +1,550 @@
+
+        String[] ownerAndRepo = UrlUtil.urlToOwnerAndReponame(mockRepo.getUrl());
+        given(githubRepoMapper.toGithubRepoGetResponse(mockRepo, ownerAndRepo[0], ownerAndRepo[1],
+                1L)).willReturn(
+                new GithubRepoGetResponse(
+                        mockUser.getId(),
+                        mockRepo.getId(),
+                        mockRepo.getName(),
+                        mockRepo.getUrl(),
+                        mockRepo.getEndDate(),
+                        ownerAndRepo[0],
+                        ownerAndRepo[1]
+                )
+        );
+
+        // when
+        GithubRepoGetResponse result = githubRepoService.getGithubRepoInfo(1L, 1L);
+
+        // then
+        assertThat(result.name()).isEqualTo(mockRepo.getName());
+        assertThat(result.userId()).isEqualTo(mockRepo.getUser().getId());
+        assertThat(result.url()).isEqualTo(mockRepo.getUrl());
+    }
+
+    @Test
+    @DisplayName("레포를 삭제할 수 있다")
+    void deleteRepo() {
+        // given
+        given(githubRepoQuery.findByUserIdAndRepositoryId(1L, 1L)).willReturn(mockRepo);
+
+        // when
+        githubRepoService.deleteRepo(1L, 1L);
+
+        // then
+        verify(contributorRepository).deleteAllByGithubRepo(mockRepo);
+        verify(notificationAgreementRepository).deleteByGithubRepoId(mockRepo.getId());
+        verify(githubRepoRepository).delete(mockRepo);
+    }
+
+    @Test
+    @DisplayName("레포를 생성할 때 레포 이름이 없으면 예외를 던진다")
+    void throwsExceptionWhenCreatingRepoWithoutName() {
+        // given
+        final String name = "";
+        final String url = "https://github.com/owner/reponame";
+        final LocalDate endDate = LocalDate.now();
+
+        GithubRepoCreateRequest request = new GithubRepoCreateRequest(
+                name,
+                url,
+                endDate,
+                Set.of("test@test.com")
+        );
+
+        // when & then
+        assertThatThrownBy(() -> githubRepoService.createGithubRepo(1L, request))
+                .isInstanceOf(GeneralException.class)
+                .hasMessageContaining(GithubRepoException.NOT_FOUND_GITHUBREPO_NAME.getMessage());
+    }
+
+    @Test
+    @DisplayName("레포를 생성할 때 url형식이 맞지 않으면 예외를 던진다")
+    void throwsExceptionWhenCreatingRepoInvalidUrl() {
+        // given
+        final String name = "repoName";
+        final String url = "https://github.com/owner";
+        final LocalDate endDate = LocalDate.now();
+
+        GithubRepoCreateRequest request = new GithubRepoCreateRequest(
+                name,
+                url,
+                endDate,
+                Set.of("test@test.com")
+        );
+
+        // when & then
+        assertThatThrownBy(() -> githubRepoService.createGithubRepo(1L, request))
+                .isInstanceOf(GeneralException.class)
+                .hasMessageContaining(GithubRepoException.INVALID_GITHUBREPO_URL.getMessage());
```

Comment:
```suggestion
        assertThatThrownBy(() -> githubRepoService.createGithubRepo(1L, request))
                .isInstanceOf(GeneralException.class)
                .extracting("exception")
                .isEqualTo(GithubRepoException.INVALID_GITHUBREPO_URL);
```
메시지로 확인할 수도 있지만 직접 예외 ENUM을 비교하여 발생하는 예외를 직접 표현하는 방법도 있습니다. 검토해보시고 필요 시 수정 부탁드립니다.

Reply:
현재는 메시지를 기준으로 예외를 확인하고 있는데, 이는 실제 사용자에게 노출되는 메시지의 정확성을 검증하고자 하는 의도가 있었습니다.

다만 말씀주신 것처럼 GeneralException 내부의 enum 값을 직접 비교하면 예외 정책을 보다 명확하게 검증할 수 있고, 메시지 변경에도 영향을 덜 받는 장점이 있다는 점에서 민재님의 방법도 고려해보겠습니다!